utilities/map.metta

```diff
@@ -39,6 +39,24 @@
 (= (Map.remove $key (ConsMap ($curKey $curVal) $tail)) 
     (if (== $key $curKey) $tail (ConsMap ($curKey $curVal) (Map.remove $key $tail))))
 
+;; Get the all the values from the map
+(: Map.values (-> (Map ($v $v)) Expression))
+(= (Map.values NilMap) ())
+(= (Map.values (ConsMap ($key $value) $tail))
+    (let $rest (Map.values $tail) (cons-atom $value $rest)))
+
+;; Get the all the keys from the map
+(: Map.keys (-> (Map ($v $v)) Expression))
```

Comment:
Same, here, instead of returning `Expression` We need to return `(List $k)` where `$k` is the type of the keys. And also correct the type of map here as well.